mobxjs/mobx-state-tree
# Fix tsconfig and typechecking

## What does this PR do and why?

This PR does two things:
1. Updates `tsconfig.json` so we start typechecking tests, and
2. adds `@types/jest`, which was missing and noticed after (1).

## Steps to validate locally

* Check out the first commit in this PR
* `rm -rf node_modules`
* `bun install`
* `bun typecheck`

That should fail. Check out the `HEAD` of this branch and then `bun install` and typecheck again, everything should pass.
